pkg/grpc/requestinfo/requestinfo.go

```diff
@@ -335,12 +348,37 @@ func sourceAddr(ctx context.Context) net.Addr {
 	return p.Addr
 }
 
-func metadataFromHeader(header http.Header) metadata.MD {
-	md := make(metadata.MD)
-	for key, vals := range header {
-		md.Append(key, vals...)
+// WithGet adds the HeaderGetter implementation to http.Header.
+type WithGet http.Header
+
+// Get implements HeaderGetter interface.
+func (h WithGet) Get(key string) []string {
+	return http.Header(h).Values(key)
+}
+
+// GetFirst returns the first value of the header by key, or empty string.
+func GetFirst(header HeaderGetter, key string) string {
+	if header == nil {
+		return ""
+	}
+	if values := header.Get(key); len(values) > 0 {
+		return values[0]
+	}
+	return ""
+}
+
+// WithHeaderMatcher wraps a header map and implements HeaderGetter interface
+// that queries the map ignoring gRPC prefixes of the header keys, stored in
+// the map: given key 'Accept' it will query for 'grpcgateway-Accept' instead.
+type WithHeaderMatcher metadata.MD
```

Comment:
`WithHeaderMatcher` is exported and has a generic name, but actually uses a very specific header mapper from the grpc-gateway library. I'd say either make it unexported or rename for clarity, or maybe both.